src/moov/trak/mdia/minf/stbl/stsd/h264/avcc.rs

```diff
@@ -166,3 +179,85 @@ impl Atom for Avcc {
         Ok(())
     }
 }
+
+/// Returns the input `n` if it is within the provided `range`. Otherwise, it returns an
+/// [`Error::InvalidSize`] error.
+fn ok_in_range(n: u8, range: std::ops::Range<u8>) -> Result<u8> {
+    if range.contains(&n) {
+        Ok(n)
+    } else {
+        Err(Error::InvalidSize)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Cursor;
+
+    #[test]
+    fn ok_in_range_is_ok_when_in_range() {
+        assert_eq!(1, ok_in_range(1, 0..8).expect("should be Ok"));
+    }
+
+    #[test]
+    fn ok_in_range_is_err_when_out_of_range() {
+        match ok_in_range(9, 0..8) {
```

Comment:
```suggestion
        match ok_in_range(8, 0..8) {
```

Just to make sure that there is no off-by-one issue here.